spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaConsumerProperties.java

```diff
@@ -253,12 +257,32 @@ public void setDestinationIsPattern(boolean destinationIsPattern) {
 		this.destinationIsPattern = destinationIsPattern;
 	}
 
+	/**
+	 * No longer used; get properties such as this via {@link #getTopic()}.
+	 * @deprecated No longer used
+	 */
+	@Deprecated
+	@DeprecatedConfigurationProperty(reason = "Not used since 2.1.1, set properties such as this via 'topic'")
 	public KafkaAdminProperties getAdmin() {
 		return this.admin;
 	}
 
 	public void setAdmin(KafkaAdminProperties admin) {
+		// Temporary workaround to copy the admin properties to the topic one. Not needed once the 'admin' field will be removed
+		final KafkaTopicProperties kafkaTopicProperties = new KafkaTopicProperties();
+		kafkaTopicProperties.setReplicationFactor(admin.getReplicationFactor());
+		kafkaTopicProperties.setReplicasAssignments(admin.getReplicasAssignments());
+		kafkaTopicProperties.setProperties(admin.getConfiguration());
+		this.topic = kafkaTopicProperties;
 		this.admin = admin;
```

Comment:
We can eliminate the redundant `admin` field by making `KafkaAdminProperties` an (empty) subclass of the new `KafkaTopicProperties`.

Reply:
Made a non empty `KafkaAdminProperties` subclass of `KafkaTopicProperties`